pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/util/beans/SettingsPersistenceUtil.java

```diff
@@ -0,0 +1,250 @@
+                stream = new FileInputStream(file);
+                Document document = builder.parse(stream);
+                return Optional.of(document);
+            } catch (SAXException | ParserConfigurationException | IOException e) {
+                e.printStackTrace();
+            } finally {
+                IOUtils.closeQuietly(stream);
+            }
+        }
+
+        return Optional.empty();
+    }
+
+
+    /**
+     * Builds a settings model recursively for the given settings owner.
+     * The properties which have a getter tagged with {@link PersistentProperty}
+     * are retrieved for later serialisation.
+     *
+     * @param root The root of the settings owner hierarchy.
+     *
+     * @return The built model
+     */
+    // test only
+    static SimpleBeanModelNode buildSettingsModel(SettingsOwner root) {
+        SimpleBeanModelNode node = new SimpleBeanModelNode(root.getClass());
+
+        for (PropertyDescriptor d : PropertyUtils.getPropertyDescriptors(root)) {
+            if (d.getReadMethod() == null) {
+                continue;
+            }
+
+            try {
+                if (d.getReadMethod().isAnnotationPresent(PersistentSequence.class)) {
+
+                    Object val = d.getReadMethod().invoke(root);
+                    if (!Collection.class.isAssignableFrom(val.getClass())) {
+                        continue;
+                    }
+
+                    @SuppressWarnings("unchecked")
+                    Collection<SettingsOwner> values = (Collection<SettingsOwner>) val;
+
+                    BeanModelNodeSeq<SimpleBeanModelNode> seq = new BeanModelNodeSeq<>(d.getName());
+
+                    for (SettingsOwner item : values) {
+                        seq.addChild(buildSettingsModel(item));
+                    }
+
+                    node.addChild(seq);
+                } else if (d.getReadMethod().isAnnotationPresent(PersistentProperty.class)) {
+                    node.addProperty(d.getName(), d.getReadMethod().invoke(root), d.getPropertyType());
+                }
+            } catch (IllegalAccessException | InvocationTargetException e) {
+                e.printStackTrace();
+            }
+
+        }
+
+        for (SettingsOwner child : root.getChildrenSettingsNodes()) {
+            node.addChild(buildSettingsModel(child));
+        }
+
+        return node;
+    }
+
+
+    /**
+     * Restores the settings from the model into the target. Dual of
+     * {@link #buildSettingsModel(SettingsOwner)}. Traverses all the
+     * tree.
+     *
+     * @param target Object in which to restore the properties
+     * @param model  The model
+     */
+    // test only
+    static void restoreSettings(SettingsOwner target, BeanModelNode model) {
+        if (model == null) {
+            return; // possibly it wasn't saved during the previous save cycle
+        }
+
+        if (target == null) {
+            throw new IllegalArgumentException();
```

Comment:
Maybe Objects.requireNonNull(target); instead? Well - it would throw a NullPointer, but you can add a message as an additional parameter.

Reply:
Yes you're right